acvm/src/compiler/transformers/csat.rs

```diff
@@ -45,6 +80,7 @@ impl CSatTransformer {
         let mut gate =
             self.partial_gate_scan_optimization(gate, intermediate_variables, num_witness);
         gate.sort();
+        self.solvable_expression(&gate);
```

Comment:
I'm not a fan of this usage of `solvable_expression`, it returning a bool implies that we're going to be handling the case where the expression is unsolvable but we don't.

Reply:
I have renamed it and it does not return a bool anymore.